src/sage/libs/gap: move a few tests to pytest

orlitzky opened the pull request:
There are two files, `src/sage/libs/gap/test{,_long}.py` that contain only tests for `sage.libs.gap`. These are doctests for no other reason than because we had no other way to run them in the past. This commit moves them to a new file, `gap_test.py`, that will be run by pytest rather than the doctest runner.

In the process I've tried to explain what I think the tests are doing.

orlitzky committed:
src/sage/libs/gap: move a few tests to pytest

There are two files, src/sage/libs/gap/test{,_long}.py that contain
only tests for sage.libs.gap. These are doctests for no other reason
than because we had no other way to run them in the past. This commit
moves them to a new file, gap_test.py, that will be run by pytest
rather than the doctest runner.

In the process I've tried to explain what I think the tests are doing.

user202729 commented on src/sage/libs/gap/gap_test.py:
```diff
@@ -0,0 +1,98 @@
+
+
+@pytest.fixture
+def tmpfile():
```
is there nothing existing like this in pytest/sage? This feels oddly general.

orlitzky replied:
Pytest has a built-in `tmpdir` that does almost the same thing, but with a directory instead of a file. But considering https://github.com/pytest-dev/pytest/issues/13669, this is safer for now.

user202729 replied:
is there any other place fixtures such as this can be placed (conftest?) We will have to eventually move it there once the same fixture is needed in another module's test anyway.

orlitzky replied:
`conftest.py` works, done.

user202729 commented on src/sage/libs/gap/gap_test.py:
```diff
@@ -0,0 +1,98 @@
+
+def test_gc_loop_1():
+    r"""
+    First stress test for garbage collection in libgap.
```
I feel "[[first]] stress test" is highly redundant, mostly because there shouldn't be any need to modify the docstring if we want to reorder the tests or add something inbetween.

orlitzky replied:
I deleted "First", "Second", and "Third" from the docstrings. There's still a number in the function name, but I don't know what else to call them.

user202729 commented on src/sage/libs/gap/gap_test.py:
```diff
@@ -0,0 +1,98 @@
+    result = True
+    for i in range(300000):
+        n = libgap.Order(H1)
+        result |= (n == two)
```
Surely you mean `&=`. Still, it feels like bad practice to use bitwise operator for boolean. What about just `assert n == two` each time here?

orlitzky replied:
I did that first but then `pytest -v` prints `OK` three hundred thousand times.

orlitzky replied:
(The `&=` is fixed, thanks.)

user202729 commented on src/sage/libs/gap/gap_test.py:
```diff
@@ -0,0 +1,98 @@
+        lis.Add(a ** 2)
+        lis.Add(b ** 2)
+        lis.Add(b * a)
+    assert True
```
isn't this assert useless?

orlitzky replied:
The original test did essentially the same thing. I can only assume that it's an "it doesn't segfault" test.

orlitzky committed:
conftest.py: factor out tmpfile fixture from sage.libs.gap

A temporary file that is deleted after the test has completed will be
useful elsewhere.

orlitzky committed:
src/doc/en/reference/libs/index.rst: drop gap/test and gap/test_long

These have been moved to pytest (gap/gap_test.py).
